custom_components/localtuya/cover.py

```diff
@@ -180,108 +195,63 @@ def __init__(
         self,
         device,
         config_entry,
-        switchid,
+        coverid,
         **kwargs,
     ):
-        #_LOGGER.info("running def __init__ of LocaltuyaCover(CoverEntity) with self=%s device=%s name=%s friendly_name=%s icon=%s switchid=%s open_cmd=%s close_cmd=%s stop_cmd=%s", self, device, name, friendly_name, icon, switchid, open_cmd, close_cmd, stop_cmd)
-        super().__init__(device, config_entry, switchid, **kwargs)
-        self._state = None
-        self._position = 50
+        super().__init__(device, config_entry, coverid, **kwargs)
+        self._position = None
+        self._current_cover_position = None
+        self._last_movement = None
+        self._last_position_set = None
+        self._last_command = None
         print(
-            "Initialized tuya cover  [{}] with switch status [{}] and state [{}]".format(
-                self.name, self._status, self._state
+            "Initialized tuya cover  [{}] with switch status [{}]".format(
+                self.name, self._status
             )
         )
 
     @property
-    def available(self):
-        """Return if device is available or not."""
-        return self._available
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        return self._current_cover_position == 100:
 
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
-        return supported_features
+    def status_updated(self):
+        """Device status was updated."""
+        self._cached_status = self._status
+        self._last_movement = self._cached_status['dps'][str(self._config[CONF_LAST_MOVEMENT])]
```

Comment:
I thought once more about these. You can use `self.dps(...)` to fetch the value (forgot about that one for a second). So basically:

```python
self._last_movement = self.dps(str._config.get(CONF_LAST_MOVEMENT))
```
This will log a warning if the requested DP does not exist.